packages/experimental/AppearanceAdditions/ios/FRNAppearanceAdditions.m

```diff
@@ -103,11 +97,23 @@ + (BOOL)requiresMainQueueSetup {
     return @[ @"appearanceChanged" ];
 }
 
+- (dispatch_queue_t)methodQueue
+{
+    return dispatch_get_main_queue();
+}
+
 - (void)startObserving {
     _hasListeners = YES;
-    _horizontalSizeClass = RCTHorizontalSizeClassPreference(nil);
-    _userInterfaceLevel = RCTUserInterfaceLevelPreference(nil);
-    _accessibilityContrastOption = RCTAccessibilityContrastPreference(nil);
+    
+    // [UITraitCollection currentTraitCollection] never returns the correct trait collection, presumably because FRNAppearanceAdditions isn't a view
+    // so it never gets updated with the right traitCollection (which happens when a view gets added to the view hierachy). In order
+    // to get the right trait collection we need to access the root view.
+    UITraitCollection *rootViewTraitCollection = [[[[[UIApplication sharedApplication] delegate] window] rootViewController] traitCollection];
```

Comment:
we can't rely on uiapplication sharedapplication. this will not work reliably with multiwindow in in ipad or external screens.

Reply:
Thanks for pointing that out, I think @mischreiber actually also mentioned to me at some point to consider multiwindow. Will look into a better way of getting the right rootviewcontroller that works when there are multiple windows

Reply:
I replaced `[UIApplication sharedApplication]` with something I found in `RCTUtils`, that we're also using the datepicker (`RCTPresentedViewController`). Here is a snippet of code from `RCTUtils` that describes what `RCTPresentedViewController` is returning:


```objective-c
UIApplication *__nullable RCTSharedApplication(void)
{
  if (RCTRunningInAppExtension()) {
    return nil;
  }
  return [[UIApplication class] performSelector:@selector(sharedApplication)];
}

UIWindow *__nullable RCTKeyWindow(void)
{
  if (RCTRunningInAppExtension()) {
    return nil;
  }

  // TODO: replace with a more robust solution
  for (UIWindow *window in RCTSharedApplication().windows) {
    if (window.keyWindow) {
      return window;
    }
  }
  return nil;
}

UIViewController *__nullable RCTPresentedViewController(void)
{
  if ([RCTUtilsUIOverride hasPresentedViewController]) {
    return [RCTUtilsUIOverride presentedViewController];
  }

  UIViewController *controller = RCTKeyWindow().rootViewController;
  UIViewController *presentedController = controller.presentedViewController;
  while (presentedController && ![presentedController isBeingDismissed]) {
    controller = presentedController;
    presentedController = controller.presentedViewController;
  }

  return controller;
}
```

I tried turning on the iPad multiwindow setting locally, and while the tester generally works as expected, there are some situations that don't behave as expected. Ex. can't have a single pane fluenttester instance on one side, and a double fluenttester instance on the other


https://user-images.githubusercontent.com/78454019/218613360-607e988d-ca41-4203-bb21-e42a22c09ffc.mov

I think supporting that scenario and others (ex. dark theme on one side, dark elevated on the other) will require a lot more thought that's beyond the scope of this PR. Also the issue seen above isn't a regression introduced by this PR, this has always been the behaviour (even before FRNAppearanceAdditions was added)

Reply:
Yeah, I agree that this is not the *final* word on these changes, but @lyzhan7's changes represent a significant improvement to the status quo.